std/uni.d

```diff
@@ -6008,6 +6008,533 @@ template SetSearcher(alias table, string kind)
     }
 }
 
+// Characters that need escaping in string posed as regular expressions
+package alias Escapables = AliasSeq!('[', ']', '\\', '^', '$', '.', '|', '?', ',', '-',
+    ';', ':', '#', '&', '%', '/', '<', '>', '`',  '*', '+', '(', ')', '{', '}',  '~');
+
+package @trusted auto memoizeExpr(string expr)()
+{
+    if (__ctfe)
+        return mixin(expr);
+    alias T = typeof(mixin(expr));
+    static T slot;
+    static bool initialized;
+    if (!initialized)
+    {
+        slot =  mixin(expr);
+        initialized = true;
+    }
+    return slot;
+}
+
+//property for \w character class
+package @property @safe CodepointSet wordCharacter()
+{
+    return memoizeExpr!("unicode.Alphabetic | unicode.Mn | unicode.Mc
+        | unicode.Me | unicode.Nd | unicode.Pc")();
+}
+
+//basic stack, just in case it gets used anywhere else then Parser
+package struct Stack(T)
+{
+@safe:
+    T[] data;
+    @property bool empty(){ return data.empty; }
+
+    @property size_t length(){ return data.length; }
+
+    void push(T val){ data ~= val;  }
+
+    @trusted T pop()
+    {
+        assert(!empty);
+        auto val = data[$ - 1];
+        data = data[0 .. $ - 1];
+        if (!__ctfe)
+            cast(void) data.assumeSafeAppend();
+        return val;
+    }
+
+    @property ref T top()
+    {
+        assert(!empty);
+        return data[$ - 1];
+    }
+}
+
+//test if a given string starts with hex number of maxDigit that's a valid codepoint
+//returns it's value and skips these maxDigit chars on success, throws on failure
+package dchar parseUniHex(Char)(ref Char[] str, size_t maxDigit)
+{
+    //std.conv.parse is both @system and bogus
+    enforce(str.length >= maxDigit,"incomplete escape sequence");
+    uint val;
+    for (int k = 0; k < maxDigit; k++)
+    {
+        immutable current = str[k];//accepts ascii only, so it's OK to index directly
+        if ('0' <= current && current <= '9')
+            val = val * 16 + current - '0';
+        else if ('a' <= current && current <= 'f')
+            val = val * 16 + current -'a' + 10;
+        else if ('A' <= current && current <= 'F')
+            val = val * 16 + current - 'A' + 10;
+        else
+            throw new Exception("invalid escape sequence");
+    }
+    enforce(val <= 0x10FFFF, "invalid codepoint");
+    str = str[maxDigit..$];
+    return val;
+}
+
+@system unittest //BUG canFind is system
```

Comment:
While it's being moved, I think it's worth noting that `canFind` is no longer system (see its [`@safe` unittest](https://github.com/dlang/phobos/blob/master/std/algorithm/searching.d#L2524)), so this can be changed.